### DISCUSSION TITLE
How to construct an OffsetDateTime from ymd+hms

### DISCUSSION THREAD

**Discussion by nicholasbishop** [time-rs/time#442] [Question] [answered]
Hi, I'm trying to figure out how to make an `OffsetDateTime` from ymd and hms values. Basically the equivalent of this in `chrono`: `Utc.ymd(2010, 1, 2).and_hms(3, 4, 5)`.

I see I can construct the two parts separately like `Date::from_calendar_date(2010, Month::January, 2)` and `UtcOffset::from_hms(3, 4, 5)`, but didn't find a constructor or conversion function that would allow them to be combined into an `OffsetDateTime`.

**Comment by jhpratt** [comment, marked as answer]
`UtcOffset::from_hms(3, 4, 5)` creates an offset from UTC, not a time.

What you're looking for is:

```rust
Date::from_calendar_date(2010, Month::January, 2)?
    .with_hms(3, 4, 5)?
    .assume_utc()
```

Assuming you want UTC to be the offset. If not, you'll want to use `PrimitiveDateTime::assume_offset`.

If these values are statically known, I suggest you use the macros instead:

```rust
datetime!(2010-01-02 03:04:05 UTC)
```

**Reply by nicholasbishop** [reply]
Perfect, thanks!
